llvm/lib/ProfileData/InstrProfCorrelator.cpp

```diff
@@ -113,11 +115,36 @@ InstrProfCorrelator::get(StringRef Filename, ProfCorrelatorKind FileKind) {
     return get(std::move(*BufferOrErr), FileKind);
   }
   if (FileKind == BINARY) {
-    auto BufferOrErr = errorOrToExpected(MemoryBuffer::getFile(Filename));
-    if (auto Err = BufferOrErr.takeError())
-      return std::move(Err);
+    if (!Filename.empty()) {
+      auto BufferOrErr = errorOrToExpected(MemoryBuffer::getFile(Filename));
+      if (auto Err = BufferOrErr.takeError())
+        return std::move(Err);
+      return get(std::move(*BufferOrErr), FileKind);
+    } else if (BIDFetcher) {
+      if (BIs->size() > 1)
```

Comment:
Do you plan to add support for multiple build IDs in a single raw profile in the future? Like a lightweight raw profile contains two build ids, one for main executable and another one for a shared library. Ideally, BIDFetcher should find/download both the executable and shared library and the correlator will match them based on their build IDs.

Reply:
The current limitation of the correlator is that it only supports correlating with a single file. When we have the support of correlating with multiple files, it should be easy to reuse that to support to correlate with multiple build IDs.

Reply:
For your reference, if you plan to add this support, I had a pending PR to add support for this feature: https://github.com/llvm/llvm-project/pull/75957.